integration/yarn/src/main/java/tachyon/yarn/ApplicationMaster.java

```diff
@@ -265,17 +309,36 @@ private void launchTachyonMasterContainers(List<Container> containers) {
 
   private void launchTachyonWorkerContainers(List<Container> containers) {
     final String command =
-        new CommandBuilder(PathUtils.concatPath(mTachyonHome, "integration", "bin",
-            "tachyon-worker-yarn.sh"))
+        new CommandBuilder("./tachyon-yarn-setup.sh").addArg("worker")
             .addArg("1>" + ApplicationConstants.LOG_DIR_EXPANSION_VAR + "/stdout")
             .addArg("2>" + ApplicationConstants.LOG_DIR_EXPANSION_VAR + "/stderr").toString();
 
     List<String> commands = Lists.newArrayList(command);
-    Map<String, String> environmentMap = new HashMap<String, String>();
-    environmentMap.put("TACHYON_MASTER_ADDRESS", mMasterContainerNetAddress);
-    environmentMap.put("TACHYON_WORKER_MEMORY_SIZE",
+
+    // Setup the environment needed for the launch context.
+    // Because our jars are available as local resources in the working directory from which
+    // the command will be run, we need to append "." to the path.
+    Map<String, String> env = new HashMap<String, String>();
```

Comment:
the setup for master and worker does similar things ... consider creating a function that input the mode (master vs. worker) and returns an environment